src/lavinmq/clustering/server.cr

```diff
@@ -161,14 +173,18 @@ module LavinMQ
           Log.error { "Disconnecting follower with the clustering id of the leader" }
           return
         end
-        if stale_follower = @followers.find { |f| f.id == follower.id }
-          Log.error { "Disconnecting stale follower with id #{follower.id.to_s(36)}" }
-          stale_follower.close
+        @lock.synchronize do
+          if stale_follower = @followers.find { |f| f.id == follower.id }
+            Log.error { "Disconnecting stale follower with id #{follower.id.to_s(36)}" }
+            @followers.delete(stale_follower)
+            stale_follower.close
+          end
+          @followers << follower # Starts in Syncing state
         end
         follower.full_sync # sync the bulk
         @lock.synchronize do
-          follower.full_sync # sync the last
-          @followers << follower
+          follower.full_sync    # sync the last
+          follower.mark_synced! # Change state to Synced
```

Comment:
Wouldn't it make more sense if `#full_sync` changes state when it's done?

Reply:
Maybe, but we do `full_sync` twice, once for bulk syncing and then again to sync the last.